### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by MichalPlachyFramgia**

### CODE REVIEW

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      render "new"
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if params[:user][:password].empty?
+      flash.now[:danger] = t "pass-empty"
```
Comment (phamkykhoi) [inline app/controllers/password_resets_controller.rb]: flash.now ? (What )
use only flash

File: app/models/user.rb [hunk 2]
```diff
@@ -60,4 +60,17 @@ def create_activation_digest
...
+    self.reset_token = User.new_token
+    update_attributes reset_digest: User.digest(reset_token), reset_sent_at: Time.zone.now
+  end
+
+  def send_password_reset_email
+    UserMailer.password_reset(self).deliver_now
+  end
+
+  def password_reset_expired?
+    reset_sent_at < 2.hours.ago
```
Comment (phamkykhoi) [inline app/models/user.rb]: Shoule define 2 is CONST

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    elsif @user.update_attributes user_params
+      log_in @user
+      flash[:success] = t "pass-reset"
+      redirect_to @user
+    end
+    render "edit"
+  end
+
+  private
+    def user_params
```
Comment (phamkykhoi) [inline app/controllers/password_resets_controller.rb]: Indent

File: app/controllers/password_resets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      params.require(:user).permit :password, :password_confirmation
+    end
+
+    def get_user
```
Comment (phamkykhoi) [inline app/controllers/password_resets_controller.rb]: Indent

File: app/controllers/password_resets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      @user = User.find_by email: params[:email]
+    end
+
+    def check_expiration
```
Comment (phamkykhoi) [inline app/controllers/password_resets_controller.rb]: Indent

File: app/controllers/password_resets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      flash.now[:danger] = t "reset-dont-send"
+      render "new"
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if params[:user][:password].empty?
```
Comment (phamkykhoi) [inline app/controllers/password_resets_controller.rb]: Not should

File: config/routes.rb [hunk 7]
```diff
@@ -9,4 +9,5 @@
 
   resources :users
   resources :account_activations, only: :edit
+  resources :password_resets, only: [:new, :create, :edit, :update]
```
Comment (phamkykhoi) [inline config/routes.rb]: Use except: [:destroy, :index]

File: app/mailers/user_mailer.rb [hunk 8]
```diff
@@ -3,4 +3,9 @@ def account_activation user
     @user = user
     mail to: user.email, subject: t("activation")
   end
+
+  def password_reset user 
+    @user = user
+    mail to: user.email, subject: t("reset")
```
Comment (phamkykhoi) [inline app/mailers/user_mailer.rb]: Remove ( )

Comment (phamkykhoi) [inline app/mailers/user_mailer.rb]: Remove ()

File: app/mailers/user_mailer.rb [hunk 9]
```diff
@@ -3,4 +3,9 @@ def account_activation user
     @user = user
     mail to: user.email, subject: t("activation")
```
Comment (phamkykhoi) [inline app/mailers/user_mailer.rb]: Remove

File: app/mailers/user_mailer.rb (hunk 8 above)
Comment (MichalPlachyFramgia) [inline app/mailers/user_mailer.rb]: app didnt work when remove ()

File: app/mailers/user_mailer.rb (hunk 9 above)
Comment (MichalPlachyFramgia) [inline app/mailers/user_mailer.rb]: app didnt work when remove ()

File: config/environment.rb [hunk 10]
```diff
@@ -15,5 +15,7 @@
   }  
 end
 
+TWOHOUR = 2
```
Comment (dieunb) [inline config/environment.rb]: Don;t push constant at here

**Comment by dieunb** [comment]
OK

**Pull request merged**